core/src/main/java/org/jruby/RubySymbol.java

```diff
@@ -150,15 +153,57 @@ public final String asJavaString() {
         return symbol;
     }
 
+    /**
+     * Return an id string (e.g. raw ISO-8859_1 charset String) for use with our method tables etc.
+     */
+    public String idString() {
+        return symbol;
+    }
+
     @Override
     public final String toString() {
-        return symbol;
+        // FIXME: This is not efficient but in theory this is just for display purposes
+        // which generally are for errors?
+        return StringSupport.byteListAsString(symbolBytes);
     }
 
-    final ByteList getBytes() {
+    public final ByteList getBytes() {
         return symbolBytes;
     }
 
+    /**
+     * Make an instance variable out of this symbol (e.g. :foo will generate :foo=).
+     * @return the new symbol
+     */
+    public RubySymbol asAccessor() {
```

Comment:
did you mean `asWriter` (not having `asReader` seems as just a coincidence since its the same symbol) ?

Reply:
Ah nice.  You're right I should have called this asWriter().  I will change that.